internal/rcm/api.go

```diff
@@ -0,0 +1,148 @@
+func (api *API) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	if api.logger != nil {
+		log.Println(request.Method, request.URL.Path)
+	}
+
+	writer.Header().Set("Content-Type", "application/json; charset=utf-8")
+	api.router.ServeHTTP(writer, request)
+}
+
+func methodNotAllowedHandler(writer http.ResponseWriter, request *http.Request) {
+	writer.WriteHeader(http.StatusMethodNotAllowed)
+}
+
+func notFoundHandler(writer http.ResponseWriter, request *http.Request) {
+	writer.WriteHeader(http.StatusNotFound)
+}
+
+func (api *API) submit(writer http.ResponseWriter, request *http.Request, _ httprouter.Params) {
+	// Check some basic HTTP parameters
+	contentType := request.Header["Content-Type"]
+	if len(contentType) != 1 || contentType[0] != "application/json" {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// JSON structure expected from the client
+	type ComposeRequest struct {
+		Distribution  string   `json:"distribution"`
+		Architectures []string `json:"architectures"`
+		Repositories  []string `json:"repositories"`
+	}
+	// Parse and verify the structure
+	var composeRequest ComposeRequest
+	decoder := json.NewDecoder(request.Body)
+	decoder.DisallowUnknownFields()
+	err := decoder.Decode(&composeRequest)
+	if err != nil || composeRequest.Distribution == "" || len(composeRequest.Architectures) == 0 || len(composeRequest.Repositories) == 0 {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Push the requested compose to the store
+	composeUUID := uuid.New()
+	err = api.store.PushCompose(composeUUID, &blueprint.Blueprint{}, make(map[string]string), "test_arch", "test_output", &target.Target{})
+	if err != nil {
+		if api.logger != nil {
+			api.logger.Println("RCM API failed to push compose:", err)
+		}
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Create the response JSON structure
+	var reply struct {
+		UUID uuid.UUID `json:"compose_id"`
+	}
+	reply.UUID = composeUUID
+	json.NewEncoder(writer).Encode(reply)
+}
+
+func (api *API) status(writer http.ResponseWriter, request *http.Request, params httprouter.Params) {
+	// JSON structure in case of error
+	var errorReason struct {
+		Error string `json:"error_reason"`
+	}
+	// Check that the input is a valid UUID
+	uuidParam := params.ByName("uuid")
+	id, err := uuid.Parse(uuidParam)
+	if err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		errorReason.Error = "Malformed UUID"
+		json.NewEncoder(writer).Encode(errorReason)
+		return
+	}
+
+	// Check that the compose exists
+	compose, exists := api.store.GetCompose(id)
+	if !exists {
+		writer.WriteHeader(http.StatusBadRequest)
+		errorReason.Error = "Compose UUID does not exist"
+		json.NewEncoder(writer).Encode(errorReason)
+		return
+	}
+
+	// JSON structure with success response
+	var reply struct {
+		Status string `json:"status"`
+	}
+
+	reply.Status = compose.QueueStatus
```

Comment:
I agree with this, and this is what we discussed. But also thinking: we may want to make the `QueueStatus` a struct with per-job information. Just like you now got per-arch information in koji when building an RPM for several architectures (not for now, just something to think about for the future).

Reply:
Yeah, definitely. I decided to go this way, because we need the changes in store, but I could leave a note in the code.